Feature/duet song/pages UI

hj9023 opened the pull request:
## 🛠️ 작업내용
- 듀엣 트레이닝 UI 구현
- 닉네임 불러오기 
- youtube link 필드에 추가 
<!-- 무엇을 변경하거나 추가했는지 설명해주세요. -->

## 📜 이슈번호
Closes #91 
closes #90 
closes #85 
<!-- #이슈번호를 작성해주세요. -->

## 📢 전달사항
#현재 듀트 post 로직#
1. 듀트 && 포스트이면: 
2. 보분리 생성 후
3. 무조건 merge 요청을 보낸다
3-1. 머지에 성공 
  3-1.1) 방 삭제 (이건 임시로. 다른 분 의견도 궁금합니다.)
  3-2.2) 머지 결과를 보분리 페이지에 포함: 머지 오디오를 앨범 커버의 재생 버튼을 통해 들을 수 있음 / 보분리 하단에 머지 오디오로 llm 생성한 것 보여줌   

3-2. 머지에 실패
  3-2.1) 기존 post 보분리만 존재
   
### 듀엣홈페이지 ###
  - 세션 시작 전
  
<img width=256  src="https://github.com/user-attachments/assets/9dc43743-e4d3-418e-8a37-c10854fd5e91">
<img width=256 src="https://github.com/user-attachments/assets/c2246e81-e514-4361-804c-c307eaa5474e">
<img width=256 src="https://github.com/user-attachments/assets/c2704e7b-1392-4f93-aff9-753af58e6322">
<img width=256 src="https://github.com/user-attachments/assets/3aa1f5fd-ec4c-4c53-a2de-bc6dd0d362a4">



   - 세션 시작 후 
<img width=256 src="https://github.com/user-attachments/assets/f0c9cebe-9c97-4080-9e48-32774954ec47">




### 방 상세 페이지 ### 
  - 방장 
<img width=256 src="https://github.com/user-attachments/assets/a3516775-0b4d-43ee-bd6d-8d243d01d81f">
<br>


  - 신청자  
<img width=256 src="https://github.com/user-attachments/assets/9502346e-98d2-4fe3-98bc-da45702be51d">

<br>

  - 뷰어 (not (방장 or 신청자))
<img width=256 src="https://github.com/user-attachments/assets/5ada61b0-a6a1-43fe-b510-17a27a61f327">



### 세팅 페이지 ###
<img width=256 src="https://github.com/user-attachments/assets/0f308d0b-da3b-4a97-9ab9-e51d449a9864">

<img width=256 src="https://github.com/user-attachments/assets/a112ebc4-f829-423d-839e-1e553efb0bbc">



### duet song list 페이지 ###
- 선택한 성부 영역에 따라 성부칩의 색이 변화됩니다.
<img width=256 src="https://github.com/user-attachments/assets/d41dc082-a465-450d-a117-87cb7ae4a581">
<img width=256 src="https://github.com/user-attachments/assets/7b5d8c31-825e-4be3-afbf-ff4069008e08">


### 연습실 만들기 - 파트 선택 ###
<img width=256 src="https://github.com/user-attachments/assets/345b50cc-6d3e-4e7f-bcd5-f770282c7474">

<img width=256 src="https://github.com/user-attachments/assets/aed7ceb8-de27-4ab1-8466-c7cded15d7ca">

<!-- PR과 관련된 내용 공유 또는 리뷰어에 대한 요청사항을 작성해주세요. -->

hj9023 committed:
✨Feat: DuetTrainingSettingPage 구현 #90

hj9023 committed:
✨Feat: 연습실(방) 및 듀엣 곡, 듀엣 신청 모델 설정  #90

hj9023 committed:
✨Feat: DuetTrainingHomePage 구현 및 라우트 수정  #90

hj9023 committed:
✨Feat: DuetRoomDetailPage, DuetSongDetailPage 구현 및 라우트 수정  #90

hj9023 committed:
✨Feat: DuetSongListPage 구현  #90

hj9023 committed:
✨Feat: SongDetailModel에 youtubeLink 추가 #91

hj9023 committed:
♻️ Refactor: DuetTrainingSettingPage 함수 인자 타입 추가 및 페이지 이동 시 안정성 확인  #90

hj9023 committed:
✨Feat: 닉네임 getter 프로바이더 추가 및 연결  #85

hj9023 committed:
✨Feat: SongDetailModel 변경에 따른 객체 생성 수정 및 목데이터 간 일관성 확보  #91

hj9023 committed:
🐞 Fix: build 안 전역 final 할당문 수정 및 가사 영역의 불필요한 스크롤반응 제거 #90

hj9023 committed:
✨Feat: API 연결 및 duetSongSection bool 인수명 변경 #90

hj9023 committed:
✨Feat: ApplicationList 스크롤로 변경 및 방주인+그외방 x 시 텍스트 추가 #90

hj9023 committed:
✨Feat: 방 생성시 홈페이지와 파트너 수락/거절 시 신청 리스트 reload #90

hj9023 committed:
🎨 Style: 불필요한 상수 및 주석 제거 #90

hj9023 commented on lib/features/vocal_analysis/views/pages/onboarding_recording_song_page.dart:
```diff
@@ -157,7 +183,7 @@ class _OnboardingRecordingSongPageState extends ConsumerState<OnboardingRecordin
                 // child: MusicContentPlayer(key: _playerKey),
                 child: MusicContentPlayer(
                   _songDetailModel,
-                  'assets/songs/datas/doremi_song_piano_v2.json',
+                  'assets/songs/datas/soda_pop_v20.json',
```
todo: 주석 제거  / json 파일 이름 수정

hj9023 commented on lib/features/vocal_analysis/views/pages/onboarding_recording_song_page.dart:
```diff
@@ -39,21 +39,47 @@ class _OnboardingRecordingSongPageState extends ConsumerState<OnboardingRecordin
     ],
     "https://sync2sing-bucket.s3.ap-northeast-2.amazonaws.com/images/album-cover/5fe33ac0-fd48-4fdb-908a-12441469fea3.jpg",
-    "assets/songs/audios/doremi_song_v3_mr.wav", // 백엔드에 저장된 파일과 음정 박자 파일 불일치 -> asset 사용. 다만 길이 조정 필요 (음악 끝나기 전까지 다음버튼 클릭 불가)
+    "assets/songs/audios/soda_pop_mr_v1.wav", // 백엔드에 저장된 파일과 음정 박자 파일 불일치 -> asset 사용. 다만 길이 조정 필요 (음악 끝나기 전까지 다음버튼 클릭 불가)
```
todo: 수정

hj9023 committed:
🐞 Fix: 듀엣홈 - 파트너 신청한 방은 '대기중인 연습실'에서 제외 #90

hj9023 committed:
🎨 Style: 불필요한 코드 및 주석 제거 #90

MinjiKim55 commented on lib/features/curriculum/views/duet_training_setting_page.dart:
```diff
@@ -366,8 +366,9 @@ class _DuetTrainingSettingPageState extends ConsumerState<DuetTrainingSettingPag
                   if (mounted) {
-                    context.pop();
-                    context.go(AppRoutePaths.duetTrainingHome);
+                    context.replace(AppRoutePaths.duetTrainingHome);
```
`context.replace(AppRoutePaths.duetTrainingHome);`
오 replace 새로 알아갑니닷

MinjiKim55 commented on lib/features/home_hub/views/before_session_widget.dart:
```diff
@@ -75,15 +74,25 @@ class _BeforeSessionWidgetState extends State<BeforeSessionWidget> {
-        for (var r in roomList) {
-          if (r.id == roomIdList[i]) {
-            partnerSentList.add(r);
+      final List<Room> pendingRooms = []; // 참ㅊ여 대기중인 연습실 x, 대기중인 연습실
```
오 좋아요`
final List<Room> pendingRooms = []; // 참ㅊ여 대기중인 연습실 x, 대기중인 연습실`
혹시 여기 주석 오타 신경쓰이시면 수정해주시면 될 것 같습니다~

MinjiKim55 replied:
필수 아님.

hj9023 replied:
이거 나중에 기억 나면 수정하겠습니다! 코드 수정할 때 계속 까먹음 －－＞ 수정 완．

MinjiKim55 reviewed (approved):
팀장님.................... 
진짜 너무 열일하신거 아닌가요 (미미코 모드 on)
항상 하게되는 말이지만, 이번엔 더더욱 고생많으셧습니다~!!!
연휴에는 푹 쉬세요!

hj9023 committed:
🐞 Fix: not beforeSession 시 room 정보 오류 수정 및 방 신청 후 홈 이동 replace 로 수정 #90

hj9023 committed:
✨Feat: 듀트 홈 afterTraining 위젯 수정 및 보분리 duet post 시 자동 병합 요청 #90
- 듀엣 레코딩 페이지 추가 및 리포트페이지에서 trainingMode에 맞는 홈으로 이동

hj9023 committed:
Merge branch 'develop' into 'feature/duet-song/pages-ui'

hj9023 committed:
🐞 Fix: 솔로 송 리스트로 이동 시의 잘못된 경로 수정

hj9023 committed:
✨Feat: 불필요한 파라미터 설정 제거 및 병합 결과(overall, feedback) 리포트에 추가
- catch -> DioException catch 절 추가해 api 통신에서 오류가 난 경우만 '수락 실패' 메시지 노출

hj9023 committed:
✨Feat: merge 성공 시 보분리에 mergeAudio 듣기 버튼 추가

hj9023 committed:
✨Feat: 보분리 목록 / 상세 조회 api 연결

hj9023 committed:
✨Feat: 듀엣 홈 당겨서 새로고침 적용 및 주석 오타 수정 #90

hj9023 committed:
🐞 Fix: 회원가입 API requestBody 수정
- gender, pitch_note_min, pitch_note_max, voice_type, age 추가
- report_id 연결 필요

hj9023 committed:
🐞 Fix: release mode 시 발생하는 오류 (인터넷 연결, 입력 안되는 문제) 해결
- 인터넷 퍼미션 추가
- login page print -> debugPrint

hj9023 committed:
🎨 Style: 불필요한 import 문 삭제
- user_profile_provider 삭제

hj9023 committed:
🎨 Style: 실행 시 debug 띠 삭제

hj9023 committed:
🐞 Fix: refresh 시 발생하는 변수 초기화 문제 (타입 불일치) 수정
- 값초기화 코드: FutureBuilder -> fetch 함수 안
- afterTraining && 커리큘럼 x 시 커리큘럼 list 호출 부분 삭제

hj9023 committed:
♻️ Refactor: textStyle 상수 사용 및 솔로 홈 UI 소폭 수정
- afterTraining && 커리큘럼 x 시 커리큘럼 list 호출 부분 삭제

hj9023 commented:
와! 이제 진짜 듀엣 브랜치 수정 끝입니다~~
변경사항 다시 확인해주시면 감사하겠습니다 만약 어려우시면 바로 머지해주시거나 코멘트 남겨주세요!

hj9023 committed:
♻️ Refactor: YoutubePlayerWidget 파라미터를 flag에서 videoStartAt 변경

 - 유튜브 플레이어 패키지 버전 업데이트 (에러문제)
 - YoutubePlayerWidget 파라미터 'flag'를 'videoStartSec'로 변경 (외부 API 이용 -> int)
- 변경된 파라미터 형식에 맞춰 기존 사용 파일들 리팩토링

hj9023 committed:
🐞 Fix: 듀엣홈 afterTraining 시 카드 연결부 UI 수정
- '트레이닝 마무리' - '훈련 카드' 연결부 중복 및 정렬 문제 수정: center -> start

hj9023 committed:
🐞 Fix: 솔로 홈 세션 카드 오버 플로우 해결
- height -> minHeight 로 수정

MinjiKim55 commented:
팀장님 쵝오!!!!!!!!!!!!!!!!!!1 시험기간에도 고생많으셨습니다~~!

일단 커밋 올라온 것까지는 모두 코드리뷰 완료했어요
혹시 아직 수정하실게  더 남으신건가여?

카드뉴스 관련 UI 는 이 브랜치 머지되고나서 만들어야 할 지 아니면 그냥 따로 할 지 고민돼서요..!

hj9023 commented:
자잘하게 고칠 게 있긴 한데 이건 시연 용으로 브랜치를 분기하는 게 좋을 것 같아요 여기 다 합치기엔 너무 많고

hj9023 committed:
🐞 Fix: room json 에서 변환 시 json 앨범 커버 이름 수정
- ['albumArtUrl'] -> ['album_art_url']
- 솔로 홈 - '트레이닝 마무리' - '훈련 카드' 연결부 중복 및 정렬 문제 수정: center -> start
